### DISCUSSION TITLE
0.25.0/how_to/process_datasets/

### DISCUSSION THREAD

**Discussion by giscus** [roboflow/supervision#1710] [Docs]
# 0.25.0/how_to/process_datasets/

A set of easy-to-use utilities that will come in handy in any computer vision project.

https://supervision.roboflow.com/0.25.0/how_to/process_datasets/

**Comment by bmorlo** [comment]
Hi!

Is there a way to load images with bounding boxes and class labels directly from .json format to a ``DetectionDataset``? If not, is there another roboflow package that might come in handy for this?

Thank you so much for your reply!
Best,
Benedikt

**Reply by LinasKo** [reply]
Hi Benedikt!

We don't have a neat solution at the moment, but there's a plan for next year to improve serialization - both with CSV, JSON and a raw format (e.g. pickle).

I'd advise unpacking the detections from json we do in [this cookbook](https://supervision.roboflow.com/develop/notebooks/serialise-detections-to-json/#convert-json-data-to-svdetections), and then constructing the dataset directly.

Note that DetectionsDataset will need a list of image paths, and a dict of `image_path` -> `detections`.

**Reply by bmorlo** [reply]
Thanks, that helped!

Best,
Benedikt
